File: java/java/src/main/java/org/signal/zkgroup/profiles/PniCredentialRequestContext.java
//
// Copyright 2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.profiles;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;
import org.signal.client.internal.Native;

public final class PniCredentialRequestContext extends ByteArray {
  public PniCredentialRequestContext(byte[] contents) throws InvalidInputException {
    super(contents);
    try {
      Native.PniCredentialRequestContext_CheckValidContents(contents);
    } catch (IllegalArgumentException e) {
      throw new InvalidInputException(e.getMessage());
    }
  }

  public ProfileKeyCredentialRequest getRequest() {
    byte[] newContents = Native.PniCredentialRequestContext_GetRequest(contents);

    try {
      return new ProfileKeyCredentialRequest(newContents);
    } catch (InvalidInputException e) {
      throw new AssertionError(e);
    }
  }
}

File: node/ts/zkgroup/profiles/PniCredentialRequestContext.ts
//
// Copyright 2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray';
import * as Native from '../../../Native';
import ProfileKeyCredentialRequest from './ProfileKeyCredentialRequest';

export default class PniCredentialRequestContext extends ByteArray {
  private readonly __type?: never;

  constructor(contents: Buffer) {
    super(contents, Native.PniCredentialRequestContext_CheckValidContents);
  }

  getRequest(): ProfileKeyCredentialRequest {
    return new ProfileKeyCredentialRequest(
      Native.PniCredentialRequestContext_GetRequest(this.contents)
    );
  }
}
